### ISSUE TITLE
bug: failed to unmarshal BigQuery spec

### ISSUE THREAD

**Issue by pandaijoeyong** [cloudquery/cloudquery#12148]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When running sudo ./cloudquery sync cloudquery-config/, i get this error with a Mysql source & Bigquery destination config. 

gcloud's application default credential folder lives in:
./config/gcloud/application_default_credentials.json

### Expected Behavior

no error

### CloudQuery (redacted) config
```yaml
kind: source
spec:
  name: "mysql"
  path: "cloudquery/mysql"
  version: "v1.2.2"
  concurrency: 10000
  tables: ["users"]
  destinations: ["bigquery"]
  spec:
    connection_string: "user:passwd@tcp(host:3306)/database"

--------------------------------------------------------------------------

kind: destination
spec:
  name: bigquery
  path: cloudquery/bigquery
  version: "v3.1.0"
  write_mode: "append"
  spec:
    project_id: project_id
    dataset_id: dataset_id
    service_account_key_json: ${file:./.config/gcloud/application_default_credentials.json}
```

### Steps To Reproduce

1. Ubuntu LTS 22.04.4 LTS
2. run  sudo ./cloudquery sync cloudquery-config/

### CloudQuery (redacted) logs
```
2023-07-13T05:05:11Z INF Loading spec(s) args=["cloudquery-config/"] module=cli
2023-07-13T05:05:11Z INF Source plugin server listening address=/tmp/cq-WjfQWrUBspDlqtaJ.sock module=cli
2023-07-13T05:05:12Z INF Plugin server listening address=/tmp/cq-bDnOvgOkyQRHBfDq.sock module=cli
2023-07-13T05:05:13Z INF started call grpc.code=OK grpc.component=server grpc.method=GetVersions grpc.method_type=unary grpc.service=cloudquery.discovery.v0.Discovery grpc.start_time=2023-07-13T05:05:13Z grpc.time_ms=0.036 module=cli peer.address=@ protocol=grpc
2023-07-13T05:05:13Z INF finished call grpc.code=OK grpc.component=server grpc.method=GetVersions grpc.method_type=unary grpc.service=cloudquery.discovery.v0.Discovery grpc.start_time=2023-07-13T05:05:13Z grpc.time_ms=0.075 module=cli peer.address=@ protocol=grpc
2023-07-13T05:05:13Z INF Start sync destinations=["bigquery (v3.1.0)"] module=cli source="mysql (v1.2.2)" sync_time=2023-07-13T05:05:13Z
2023-07-13T05:05:13Z INF started call grpc.code=OK grpc.component=server grpc.method=Init grpc.method_type=unary grpc.service=cloudquery.source.v1.Source grpc.start_time=2023-07-13T05:05:13Z grpc.time_ms=0.027 module=cli peer.address=@ protocol=grpc
2023-07-13T05:05:13Z INF finished call grpc.code=OK grpc.component=server grpc.method=Init grpc.method_type=unary grpc.service=cloudquery.source.v1.Source grpc.start_time=2023-07-13T05:05:13Z grpc.time_ms=487.82 module=cli peer.address=@ protocol=grpc
2023-07-13T05:05:13Z INF started call grpc.code=OK grpc.component=server grpc.method=GetDynamicTables grpc.method_type=unary grpc.service=cloudquery.source.v1.Source grpc.start_time=2023-07-13T05:05:13Z grpc.time_ms=0.023 module=cli peer.address=@ protocol=grpc
2023-07-13T05:05:13Z INF finished call grpc.code=OK grpc.component=server grpc.method=GetDynamicTables grpc.method_type=unary grpc.service=cloudquery.source.v1.Source grpc.start_time=2023-07-13T05:05:13Z grpc.time_ms=0.219 module=cli peer.address=@ protocol=grpc
2023-07-13T05:05:13Z INF started call grpc.code=OK grpc.component=server grpc.method=Configure grpc.method_type=unary grpc.service=proto.Destination grpc.start_time=2023-07-13T05:05:13Z grpc.time_ms=0.02 module=cli peer.address=@ protocol=grpc
2023-07-13T05:05:13Z ERR finished call grpc.code=Unknown grpc.component=server grpc.error="failed to initialize client: failed to unmarshal BigQuery spec: json: cannot unmarshal object into Go struct field Spec.service_account_key_json of type string" grpc.method=Configure grpc.method_type=unary grpc.service=proto.Destination grpc.start_time=2023-07-13T05:05:13Z grpc.time_ms=0.368 module=cli peer.address=@ protocol=grpc
2023-07-13T05:05:13Z INF End sync destinations=["bigquery (v3.1.0)"] module=cli source="mysql (v1.2.2)" sync_time=2023-07-13T05:05:13Z
2023-07-13T05:05:13Z INF Sending sync summary to analyticsv1.cloudquery.io:443 module=cli
2023-07-13T05:05:13Z INF waiting for source plugin to terminate module=cli
2023-07-13T05:05:13Z INF Got stop signal. Plugin server shutting down address=/tmp/cq-bDnOvgOkyQRHBfDq.sock module=cli signal=interrupt
2023-07-13T05:05:13Z INF waiting for source plugin to terminate module=cli
2023-07-13T05:05:13Z INF Got stop signal. Source plugin server shutting down address=/tmp/cq-WjfQWrUBspDlqtaJ.sock module=cli signal=interrupt
2023-07-13T05:05:13Z ERR exiting with error error="failed to sync v1 source mysql: rpc error: code = Unknown desc = failed to initialize client: failed to unmarshal BigQuery spec: json: cannot unmarshal object into Go struct field Spec.service_account_key_json of type string" module=cli
```

### CloudQuery version

3.9.0

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by erezrokah** [comment]
Hi @pandaijoeyong thank you for opening the issue.
Can you try the following configuration:

```yaml
kind: source
spec:
  name: "mysql"
  path: "cloudquery/mysql"
  version: "v1.2.2"
  concurrency: 10000
  tables: ["users"]
  destinations: ["bigquery"]
  spec:
    connection_string: "user:passwd@tcp(host:3306)/database"

---

kind: destination
spec:
  name: bigquery
  path: cloudquery/bigquery
  version: "v3.1.0"
  write_mode: "append"
  spec:
    project_id: project_id
    dataset_id: dataset_id
    service_account_key_json: |
      ${file:./.config/gcloud/application_default_credentials.json}
```

That should ensure the service account is treated as a raw string instead of being embedded in the YAML file as it.

**Comment by pandaijoeyong** [comment]
Hi @erezrokah thanks for the quick response!

I've tried updating my source & config files (as much as deleting & copy-pasting while updating the credentials etc), still having the unmarshal error but now it cannot find the expected key:


Here's the logs:
`2023-07-14T02:31:07Z INF Loading spec(s) args=["cloudquery-config/"] module=cli                                                             2023-07-14T02:31:07Z ERR exiting with error error="failed to load spec(s) from cloudquery-config/. Error: failed to unmarshal file cloudquery-config/bigquery_destination.yaml: failed to convert yaml to json: yaml: line 9: did not find expected key" module=cli`



Additionally the perms of my json file:
`-rwxr-xr-x 1 root   root    2377 Jul 14 02:33 application_default_credentials.json`


Additionally the structure of my json key (redacted):
```
{
  "type": "service_account",
  "project_id": "project_id",
  "private_key_id": "private_key_id",
  "private_key": "-----BEGIN PRIVATE KEY-----\nprivate\nkey\n-----END PRIVATE KEY-----\n",
  "client_email": "client_email",
  "client_id": "client_id",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "auth_provider_x509_cert_url",
  "client_x509_cert_url": "client_x509_cert_url",
  "universe_domain": "googleapis.com"
}
```

**Comment by erezrokah** [comment]
HI @pandaijoeyong, a fix to this issue is now available in `v3.10.1` of the CLI. See https://github.com/cloudquery/cloudquery/releases/tag/cli-v3.10.1.

Please comment if the issue still persists
